packages/amplify_core/lib/src/types/notifications/push/push_notification_message.dart

```diff
@@ -55,16 +62,26 @@ class PushNotificationMessage
         );
       }
     }
+    return PushNotificationMessage(
+      title: title,
+      body: body,
+      imageUrl: imageUrl,
+      deeplinkUrl: deeplinkUrl,
+      goToUrl: goToUrl,
+      apnsOptions: apnsOptions,
+      fcmOptions: fcmOptions,
+      data: data,
+    );
   }
 
-  String? title;
-  String? body;
-  String? imageUrl;
-  String? deeplinkUrl;
-  String? goToUrl;
-  FcmPlatformOptions? fcmOptions;
-  ApnsPlatformOptions? apnsOptions;
-  Map<Object?, Object?> data = {};
+  final String? title;
+  final String? body;
+  final String? imageUrl;
+  final String? deeplinkUrl;
+  final String? goToUrl;
+  final FcmPlatformOptions? fcmOptions;
+  final ApnsPlatformOptions? apnsOptions;
+  final Map<Object?, Object?> data;
```

Comment:
This will always be `Map<String, Object?>` if I'm not mistaken.
```suggestion
  final Map<String, Object?> data;
```

Reply:
The data payload from iOS comes through as `Object?` which is why I kept it as that.